packages/core/tests/repository/cache.repository.test.ts

```diff
@@ -0,0 +1,212 @@
+
+            expect(result).toBe(book);
+        });
+
+        it('should return the expected value when the key is already cached', async () => {
+            const query = getDefaultKeyQuery();
+            const operation = getCacheOperation();
+            const book = getDefaultBook();
+
+            await repository.put(book, query, operation);
+            const result = await repository.get(query, operation);
+
+            expect(result).toBe(book);
+        });
+    });
+
+    describe('put', () => {
+        it('should throw an error when a non-supported operation is given', async () => {
+            const book = getDefaultBook();
+            const keyQuery = getDefaultKeyQuery();
+            const notSupportedOperation = getNotSupportedOperation();
+
+            const result = repository.put(book, keyQuery, notSupportedOperation);
+
+            expect(result).rejects.toThrowError(OperationNotSupportedError);
+        });
+
+        it('should return the given value when this is inserted', async () => {
+            const book = getDefaultBook();
+            const keyQuery = getDefaultKeyQuery();
+            const operation = getDefaultOperation();
+
+            const result = await repository.put(book, keyQuery, operation);
+
+            expect(result).toBe(book);
+        });
+
+        it('should override the value when an already existing key is given', async () => {
+            const query = getDefaultKeyQuery();
+            const operation = getDefaultOperation();
+            const bookOne = getDefaultBook();
+            const bookTwo = getRandomBook();
+
+            await repository.put(bookOne, query, operation);
+            await repository.put(bookTwo, query, operation);
+            const result = await repository.get(query, operation);
+
+            expect(result).toBe(bookTwo);
+        });
+
+        it('should not update the cache when the MainSyncOperation fails', async () => {
+            const book = getDefaultBook();
+            const query = getNotSupportedQuery();
+            const operation = getMainSyncOperation();
+
+            const syncResult = repository.put(book, query, operation)
+            const cacheResult = repository.get(getDefaultKeyQuery(), getCacheOperation());
+
+            await expect(syncResult).rejects.toThrowError(QueryNotSupportedError);
+            await expect(cacheResult).rejects.toThrowError(NotFoundError);
+        });
+    });
+
+    describe('delete', () => {
+        it('should throw an error when a non-supported operation is given', () => {
+            const keyQuery = getDefaultKeyQuery();
+            const notSupportedOperation = getNotSupportedOperation();
+
+            const result = repository.delete(keyQuery, notSupportedOperation);
+
+            expect(result).rejects.toThrowError(OperationNotSupportedError);
+        });
+
+        it('should return a value for a given key when the delete was performed on another key', async () => {
```

Comment:
```suggestion
        it(`shouldn't affect non-related keys (deletion doesn't have side effects)`, async () => {
```

Reply:
Following what Miguel said:
My personal pick of all those options is this -> "should HAVE_EXPECTED_BEHAVIOUR when RACE_CONDITION"

Reply:
Is super weird. The `when` denotes causality, no? So… delete was performed on another key, which triggered a value return for the given key?

What about? `should not affect other cache entries when an entry is deleted`. This reads better: an entry was deleted which didn't affect other cache entries.